### PR TITLE
rm non-stdlib module `packaging`

### LINKED ISSUE #1069: ch-run-oci: stop using `distutils`
In Python 3.10, `distutils` is [deprecated](https://www.python.org/dev/peps/pep-0632/) and raises `DeprecationWarning`.

`ch-run-oci` currently uses it for version parsing.

### PR DESCRIPTION
**Pull request by reidpr**
Closes #1069.

### CODE REVIEW

**Comment by reidpr** [comment]
The cancelled check looks spurious, and internal CI passed.

File: bin/ch-run-oci.py.in [hunk 1]
```diff
@@ -320,6 +319,11 @@ def state_load():
 
    return st
 
+def version_parse_simple(s):
+   # Dead-simple version parsing function that’s good enough for OCI in this
+   # context.
+   return tuple(re.split(r"[.-]", s)[:3])
```
Comment (j-ogas) [inline bin/ch-run-oci.py.in:325]:
```suggestion
def version_parse_oci(s):
   # Dead-simple version parsing for OCI; not intended for other uses.
   return tuple(re.split(r"[.-]", s)[:3])
```

File: bin/ch-run-oci.py.in [hunk 2]
```diff
@@ -301,9 +300,9 @@ def state_load():
    st.config = json.load(fs.Path(args.bundle + "/config.json").open_("rt"))
    #debug_lines(json.dumps(st.config, indent=2))
 
-   v_min = packaging.version.Version(OCI_VERSION_MIN)
-   v_actual = packaging.version.Version(st.config["ociVersion"])
-   v_max = packaging.version.Version(OCI_VERSION_MAX)
+   v_min = version_parse_simple(OCI_VERSION_MIN)
+   v_actual = version_parse_simple(st.config["ociVersion"])
+   v_max = version_parse_simple(OCI_VERSION_MAX)
```
Comment (j-ogas) [inline bin/ch-run-oci.py.in:305]:
```suggestion
   v_min = version_parse_oci(OCI_VERSION_MIN)
   v_actual = version_parse_oci(st.config["ociVersion"])
   v_max = version_parse_oci(OCI_VERSION_MAX)
```

The method name can imply general use. The method appears to be intended for OCI version parsing only. Perhaps the method name should be a little more pointed?

**Review by j-ogas — CHANGES_REQUESTED** [review]
See comments.

**Comment by reidpr** [comment]
Failed test looks spurious, and it passed internal CI.

**Review by j-ogas — APPROVED** [review]

**Review by lucaudill — APPROVED** [review]
Looks good to me

**Pull request merged**